Orange/widgets/widget.py

```diff
@@ -738,6 +745,32 @@ def _userconfirmed():
 
         self.__msgwidget.accepted.connect(_userconfirmed)
 
+    @classmethod
+    def migrate_settings(cls, settings, version):
```

Comment:
This belongs to whoever defines the settings handler (the class passed in parameter to SettingHandler.create call). As components can be used as a part of the widget, they do not have their own settingHandlers and reuse the widget's. The widget is therefore responsible for migration of all the settings.